sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala

```diff
@@ -115,6 +115,8 @@ protected[sql] abstract class AtomicType extends DataType {
   private[sql] type InternalType
   private[sql] val tag: TypeTag[InternalType]
   private[sql] val ordering: Ordering[InternalType]
+
+  private[spark] override def supportsEquals: Boolean = true
```

Comment:
I don't think this should be a property of the data type. It's specific to the `OpenHashSet`. How about we add this method to `object OpenHashSet`?

Reply:
Not all of the expressions utilize ```OpenHashSet``` or ```OpenHashMap```. What about ```TypeUtils``` that contains methods like ```getInterpretedOrdering```?

Reply:
SGTM

Reply:
+1 too

Reply:
+1